### PR TITLE
test: Code Coverage: Personal Cards Page #4

### PR DESCRIPTION
**Pull request by jayfyle**
### <samp>🤖 Generated by Copilot at a6e0704</samp>

This pull request adds and updates unit tests and mock data for the personal cards page and service, which enable users to view and create expenses from their personal cards. It also adds optional chaining operators to the `openExpensePreview` method in `personal-cards.page.ts` to prevent errors if the transaction details are missing.

### <samp>🤖 Generated by Copilot at a6e0704</samp>

> _`Mock data` fills the void of the real_
> _Testing the page of `personal cards`_
> _We chain the operators to prevent the kill_
> _Creating expenses from the scars_

### <samp>🤖 Generated by Copilot at a6e0704</samp>

*  Add mock data objects for personal card query param filters and matched transaction ([link](https://github.com/fylein/fyle-mobile-app/pull/2403/files?diff=unified&w=0#diff-6fe7c64e8d36575aaa8d32fd7f3b48031688077375d32f158ec8639db22de0aeR28-R36), [link](https://github.com/fylein/fyle-mobile-app/pull/2403/files?diff=unified&w=0#diff-8bdff62a2bf5c5df5ecaeebef322a7d2ddb91f55fd1332748d6ce71b4efe2283R37-R79))
*  Update imports in `personal-cards.page.spec.ts` to include new mock data and Ionic components and modules ([link](https://github.com/fylein/fyle-mobile-app/pull/2403/files?diff=unified&w=0#diff-6e314ff0829ddfea53cbbca077b821775702ee8b5c5804cedab82e27a8a6f8a7L9-R25), [link](https://github.com/fylein/fyle-mobile-app/pull/2403/files?diff=unified&w=0#diff-6e314ff0829ddfea53cbbca077b821775702ee8b5c5804cedab82e27a8a6f8a7R40-R43))
*  Add unit tests for new methods and scenarios in `personal-cards.page.spec.ts`, such as clearing text, creating expense, opening expense preview and date range modal, and refreshing data ([link](https://github.com/fylein/fyle-mobile-app/pull/2403/files?diff=unified&w=0#diff-6e314ff0829ddfea53cbbca077b821775702ee8b5c5804cedab82e27a8a6f8a7R625-R793))
*  Add optional chaining operators to prevent errors in `openExpensePreview` method in `personal-cards.page.ts` ([link](https://github.com/fylein/fyle-mobile-app/pull/2403/files?diff=unified&w=0#diff-9969659182f6d9c766cc13e4b16549635f65f56db4c7f33df952fe8a1ce979aeL748-R752))

## Clickup
https://app.clickup.com/t/85ztx3nye

## Code Coverage
`78.59% Statements 235/299`
`73.46% Branches 36/49`
`71.23% Functions 52/73`
`79.05% Lines 234/296`

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

File: src/app/fyle/personal-cards/personal-cards.page.spec.ts [hunk 1]
```diff
@@ -608,6 +622,175 @@ describe('PersonalCardsPage', () => {
         expect(personalCardsService.generateFilterPills).toHaveBeenCalledTimes(2);
       });
     });
+
+    describe('clearText():', () => {
+      it('should clear text and remove focus from search bar', () => {
+        component.clearText('onSimpleSearchCancel');
+
+        expect(component.simpleSearchText).toEqual('');
+        expect(component.isSearchBarFocused).toBeTrue();
```
Comment (Chethan-Fyle) [inline src/app/fyle/personal-cards/personal-cards.page.spec.ts:631]: According to the description, shouldn't this result in false?? 🤔

File: src/app/fyle/personal-cards/personal-cards.page.spec.ts [hunk 2]
```diff
@@ -608,6 +622,175 @@ describe('PersonalCardsPage', () => {
...
+      });
+
+      it('should set search bar if not redirected from cancel', () => {
```
Comment (Chethan-Fyle) [inline src/app/fyle/personal-cards/personal-cards.page.spec.ts:634]: Description and test behaviour are contradictory.

**Review by Chethan-Fyle — APPROVED** [review]
Two p1 comments.

**Review by OmkarJ13 — APPROVED** [review]

**Review by Dimple16 — COMMENTED** [review]
PR is reviewed, will skip

**Pull request merged**